src/Middleware/WebSockets/src/WebSocketMiddleware.cs

```diff
@@ -209,17 +239,22 @@ public static bool CheckSupportedWebSocketRequest(string method, IHeaderDictiona
                 return false;
             }
 
+            if (!CheckWebSocketVersion(requestHeaders))
+            {
+                return false;
+            }
+
             var foundHeader = false;
 
-            var values = requestHeaders.GetCommaSeparatedValues(HeaderNames.SecWebSocketVersion);
+            var values = requestHeaders.GetCommaSeparatedValues(HeaderNames.Connection);
```

Comment:
https://datatracker.ietf.org/doc/html/rfc8441#section-5
> [RFC6455] requires the use of Connection and Upgrade header fields
   that are not part of HTTP/2.  They MUST NOT be included in the
   CONNECT request defined here.

Reply:
CheckSupportedWebSocketRequest only covers HTTP/1.1, CheckSupportedWebSocketRequest covers HTTP/2.